src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackDecoder.cs

```diff
@@ -265,6 +270,7 @@ public void Decode(ReadOnlySpan<byte> data, bool endHeaders, HeaderCallback onHe
                     case State.HeaderNameLengthContinue:
                         if (_integerDecoder.Decode(b))
                         {
+                            Debug.Assert(_integerDecoder.Value != 0, "HPACK integer decoder failed to stop an overlong 0.");
```

Comment:
What does "an overlong 0" mean?

Reply:
over long encoding. the integer 0 should take only a single byte, any more and `IntegerDecoder` will throw an exception.

Reply:
I'm still having trouble parsing this.  By "over long", do you mean "too long" / "too many bytes"?

Reply:
It's terminology that came to me via UTF-8 security issues and I'm realizing now that it may not be common knowledge. I will update the assert message with something else.

Overlong encoding means an integer was encoded in more bytes than was required. `IntegerDecoder` does not allow this. The assert is there mostly as documentation that we don't need to check the 0 case on continuation bytes because it is prevented elsewhere.

Reply:
I see.  Wasn't familiar with the term.  Thanks for the explanation.